Find a functionally equivalent code:android.media.MediaFocusControl.removeFocusStackEntryForClient:COMMENT
Method Modifier: private     
Comment:/**
 * Helper function:
 * Called synchronized on mAudioFocusLock
 * Remove focus listeners from the focus stack for a particular client when it has died.
 */

@@ -5,11 +5,11 @@
     // evaluated it, traversal order doesn't matter here)
     Iterator<FocusRequester> stackIterator = mFocusStack.iterator();
     while (stackIterator.hasNext()) {
-        FocusRequester fr = (FocusRequester) stackIterator.next();
+        FocusRequester fr = stackIterator.next();
         if (fr.hasSameBinder(cb)) {
             Log.i(TAG, "AudioFocus  removeFocusStackEntry(): removing entry for " + cb);
             stackIterator.remove();
@@ -16,24 +16,20 @@
         // we removed an entry at the top of the stack:
         // notify the new top of the stack it gained focus.
         notifyTopOfAudioFocusStack();
-        // there's a new top of the stack, let the remote control know
-        synchronized (mRCStack) {
-            checkUpdateRemoteControlDisplay_syncAfRcs(RC_INFO_ALL);
-        }
     }
 }

